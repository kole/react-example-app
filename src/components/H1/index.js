import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const H1 = styled.div`
    font-size: 3em;
    font-weight: bold;
    line-height: 1.3em;
`;

const H1Component = (props) => {
    const { text } = props;
    return <H1>{text}</H1>
};

H1Component.propTypes = {
    text: PropTypes.string.isRequired
};

export default H1Component;
