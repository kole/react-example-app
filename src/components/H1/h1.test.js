/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import H1 from './index.js';

describe('H1', () => {
    test('Snapshot', () => {
        const component = renderer
            .create(<H1 text="H1 Component Test" />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
