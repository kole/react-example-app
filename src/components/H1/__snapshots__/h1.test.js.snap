// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 Snapshot 1`] = `
<div
  className="sc-bdVaJa iSJzkt"
>
  H1 Component Test
</div>
`;
